### PR TITLE
Context location setup

### PR DESCRIPTION
**Pull request by norcross**
### Changes proposed in this pull request

extending the use of the `context` arg to allow for placement in other locations inside the post editor. new areas:
- **form_top** : calls the `edit_form_top` action
- **before_permalink** : calls the `edit_form_before_permalink` action
- **after_title** : calls the `edit_form_after_title` action
- **after_editor** : calls the `edit_form_after_editor` action

### CODE REVIEW

File: includes/CMB2_hookup.php [hunk 1]
```diff
@@ -294,6 +329,50 @@ public function return_column_display( $empty, $custom_column, $object_id ) {
...
+			return;
+		}
+
+		// If we aren't applying one of our extended contextes, just return our array.
+		if ( empty( $cmb->prop( 'context' ) ) || ! in_array( $cmb->prop( 'context' ), array( 'form_top', 'before_permalink', 'after_title', 'after_editor' ) ) ) {
+			return $classes;
+		}
+
+		// Include a generic context wrapper and the postbox wrapper.
+		$classes[] = 'postbox cmb2-context-wrap';
```
Comment (norcross) [inline includes/CMB2_hookup.php]: the `postbox` class ensures it looks like the other metaboxes. I can create a conditional if you'd prefer.

File: includes/CMB2_hookup.php [hunk 2]
```diff
@@ -294,6 +329,50 @@ public function return_column_display( $empty, $custom_column, $object_id ) {
...
 
 	/**
+	 * [add_context_metabox description]
+	 */
+	public function add_context_metabox() {
+
+		if ( ! $this->show_on() ) {
+			return;
+		}
+
```
Comment (norcross) [inline includes/CMB2_hookup.php]: 👍

File: includes/CMB2_hookup.php (hunk 1 above)
Comment (jtsternberg) [inline includes/CMB2_hookup.php]: @norcross makes sense. I think I was seeing it cached, so it looked a little wonky. What do you think about outputting the entire metabox markup, allowing "metaboxes" in those other locations?

**Comment by jtsternberg** [comment]
💥  thank you Mr. @norcross!

**Pull request merged**